types/underscore/index.d.ts

```diff
@@ -166,6 +166,18 @@ declare module _ {
 
     type Truthy<T> = Exclude<T, AnyFalsy>;
 
+    type PickSelector<V> = (keyof V)[] | string[] | ObjectIterator<TypeOfDictionary<V, any>, boolean, V>;
+
+    type _Pick<V, I> =
+        Extract<I, keyof V> extends never ? Partial<V>
+        : Pick<V, Extract<I, keyof V>>;
```

Comment:
How clever is TSC about memoizing the value of repeated expressions (types)? If not very clever, maybe prevent `Extract<I, keyof V>` from being evaluated twice by lifting it into a type parameter default:

```suggestion
    type _Pick<V, I, K = Extract<I, keyof V>> =
        K extends never ? Partial<V>
        : Pick<V, K>;
```

Reply:
Even if TS is clever enough to not need it, I like the idea of a separate K because it makes everything else easier to read (especially if I might be able to constrain it to `keyof V` to make it clearer what the result is expected to be).

Reply:
Hmm, unfortunately doing that triggers the conditional expression behavior that causes TS to iterate over the types in `K` and make a union of the results instead of evaluating `K` as a single type, so instead of `Pick<MixedTypeRecord, "a" | "b">` one ends up with `Pick<MixedTypeRecord, "a"> | Pick<MixedTypeRecord, "b">`. I want to say that TS will do that any time the left-hand side of `extends` is just a single type variable with no modification, but I'm not 100% sure.